src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,179 @@
+        SPEED("§6+600⸕", Renderable.string(""));
+
+        override fun toString() = label
+
+        companion object {
+            @JvmField
+            val defaultOption = listOf(
+                TITLE, TIMER, STREAK, SPEED
+            )
+        }
+    }
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        displayDirty = true
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) { //TODO: Remove once #2540 is merged
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+            displayDirty = true
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        attemptClearDisplay()
+    }
+
+    private fun attemptClearDisplay() {
+        if (streakEndTimer.isInFuture()) return
+        blockBreakStreak = 0
+        displayDirty = true
+        createDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (!config.flowstateHelper) return
+        if (hasFlowstate() == null) return
+
+        if (display == null || streakEndTimer.isInFuture()) createDisplay()
+        display?.let {
+            config.flowstateHelperPosition.renderRenderables(
+                listOf(Renderable.verticalContainer(it, 1)),
+                posLabel = "Flowstate Helper",
+                addToGuiManager = true
+            )
+        }
+    }
+
+    private fun createDisplay() {
+        if (displayDirty) {
+            displayDirty = false
+            createDisplayTitle()
+            createDisplayTimer()
+            createDisplayBlock()
+            createDisplaySpeed()
+            display = buildList {
+                config.flowstateHelperAppearance.forEach {
+                    add(it.renderable)
+                }
+            }
+        } else {
+            createDisplayTimer()
+            display = buildList {
+                config.flowstateHelperAppearance.forEach {
+                    add(it.renderable)
+                }
+            }
+        }
+    }
+
+    private fun createDisplayTitle(): Renderable {
+        FlowstateElements.TITLE.renderable = Renderable.string("§d§lFlowstate Helper")
+        return FlowstateElements.TITLE.renderable
+    }
```

Comment:
This can be static. Just set it in the enum once. So there is never a need to update it.